src/Umbraco.Infrastructure/Persistence/Repositories/Implement/UserRepository.cs

```diff
@@ -154,6 +154,49 @@ internal class UserRepository : EntityRepositoryBase<int, IUser>, IUserRepositor
     public IUser? GetByUsername(string username, bool includeSecurityData) =>
         GetWith(sql => sql.Where<UserDto>(x => x.Login == username), includeSecurityData);
 
+    public IUser? GetForUpgradeByUsername(string username) => GetUpgradeUserWith(sql => sql.Where<UserDto>(x => x.Login == username));
+
+    public IUser? GetForUpgradeByEmail(string email) => GetUpgradeUserWith(sql => sql.Where<UserDto>(x => x.Email == email));
```

Comment:
Are we sure this is case insensitive?

Reply:
Just double checked this, but trying to migrate with mixed case email, and it works fine.

Looking a bit into it to be sure and [MSSQL is case insensitive by default](https://stackoverflow.com/a/1224381), and looking at [SqliteSyntaxProvider](https://github.com/umbraco/Umbraco-CMS/blob/contrib/src/Umbraco.Cms.Persistence.Sqlite/Services/SqliteSyntaxProvider.cs#L60) our sqlite implementation is also defaulted to be case insensitive 😄 

But very good point :D I learned something new today 🕵️‍♂️